### ISSUE TITLE
Arrow Label Dropping to the Origin

### ISSUE THREAD

**Issue by davidf628** [jsxgraph/jsxgraph#516]
I made a JSFiddle to make sure the issue I'm explaining is clear: [https://jsfiddle.net/46zvtuj0/](https://jsfiddle.net/46zvtuj0/)

The issue is that the label for the first vector (v1) drops down to the origin as you change the angle measure with the slider, but the label for the second vector (v2) stays at the end of the vector as expected. 

Is there a way to prevent the label for vector v1 from repositioning at the origin?

**Comment by alfredwassermann** [comment]
Thanks a lot for the bug report. Could it be that the jsfiddle does not contain all the code?

**Comment by davidf628** [comment]
I originally forgot to press the save button on the jsfiddle I created, so perhaps the link I supplied sent you to an empty jsfiddle. Here's another try at that:
[https://jsfiddle.net/davidflenner/6mbxhn1q/](https://jsfiddle.net/davidflenner/6mbxhn1q/)

But just in case, I made a demo of this outside of jsfiddle that should work:
[https://davidf628.github.io/VectorTest.html](https://davidf628.github.io/VectorTest.html)

In addition, here's an animated gif of me using the app in case that's not working: 
[https://davidf628.github.io/Animation.gif](https://davidf628.github.io/Animation.gif)

My thought is that the Arrow item is choosing which end of the arrow to place the label, and initially it picks the endpoint of the arrow, but once it gets rotated a bit it then chooses the starting point of the arrow. I verified that if you continue rotating the second vector past 90 degrees it also has its label moved to the origin. So this may just be the functionality of the Arrow and maybe not a bug, but it seems like the label should stay at the endpoint. Is there an option that would allow me to keep the label anchored to the endpoint?

**Comment by alfredwassermann** [comment]
Indeed this is a bug which should be fixed. In the meantime, a workaround is to set a label to the invisible second defining point of the vector:

```
var v1 = board.create('arrow', [[0, 0], [
	() => 2 * Math.cos((s.Value() + 90) * Math.PI / 180),
        () => 2 * Math.sin((s.Value() + 90) * Math.PI / 180)
  ]], {point2: {withLabel: true, name: 'T_1', label: {visible: true}}});
```

**Comment by alfredwassermann** [comment]
Fixed by d8b0767bfcee93f960b840c897c6069783f4517d
